easybuild/tools/modules.py

```diff
@@ -520,14 +521,14 @@ def run_module(self, *args, **kwargs):
                 out = "stdout: %s, stderr: %s" % (stdout, stderr)
                 raise EasyBuildError("Changing environment as dictated by module failed: %s (%s)" % (err, out))
 
-            # correct LD_LIBRARY_PATH as yielded by the adjustments made
+            # correct $LD_LIBRARY_PATH and $LD_PRELOAD as yielded by the adjustments made
             # make sure we get the order right (reverse lists with [::-1])
-            curr_ld_library_path = os.environ.get('LD_LIBRARY_PATH', '').split(':')
-            new_ld_library_path = [x for x in nub(prev_ld_library_path + curr_ld_library_path[::-1]) if len(x)][::-1]
+            for key in LD_ENV_VAR_KEYS:
+                curr_ld_val = os.environ.get(key, '').split(os.pathsep)
+                new_ld_val = [x for x in nub(prev_ld_values[key] + curr_ld_val[::-1]) if len(x)][::-1]
```

Comment:
I think we need the len there to check for occurences of False or None, the len will only be False on the empty string, so I'm not sure if you can just remove it.